### PR TITLE
Plans and Subscriptions Template CRUD - READ

### PR DESCRIPTION
**Pull request by ianlessa**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Issues | mundipagg/plug-team#306
| What?         | Plans and Subscriptions Template CRUD - READ
| Why?          | To read and show on frontend what was created
| How?          | Twig edits and repository changes.

#### Screen
screenshot from 2018-07-05 11-00-46

#### ER Diagram

screenshot from 2018-07-04 18-12-36

#### Class Diagram

screenshot from 2018-07-05 10-18-37

### CODE REVIEW

**Comment by ianlessa** [comment]
Modificar campo description da tabela template para tipo texto

**Comment by ianlessa** [comment]
Remover o input de dias do vencimento, quando o tipo for Pré pago ou pós pago.

**Comment by ianlessa** [comment]
conferir nivel de acesso dos atributos das classes nos diagrams

**Comment by ianlessa** [comment]
no javascript trocar "cicle" por "cycle"

File: system/library/mundipagg/src/Aggregates/Template/DueValueObject.php [hunk 1]
```diff
@@ -2,12 +2,14 @@
 
 namespace Mundipagg\Aggregates\Template;
 
+
```
Comment (michelpl) [inline system/library/mundipagg/src/Aggregates/Template/DueValueObject.php]: Linha extra

File: system/library/mundipagg/src/Aggregates/Template/DueValueObject.php [hunk 2]
```diff
@@ -2,12 +2,14 @@
...
 use Exception;
 
 class DueValueObject
 {
-    const TYPE_EXACT = 'E';
-    const TYPE_WORKDAY = 'U';
+    const TYPE_EXACT = 'X';
+    const TYPE_PREPAID = 'E';
+    const TYPE_POSPAID = 'O';
```
Comment (michelpl) [inline system/library/mundipagg/src/Aggregates/Template/DueValueObject.php]: Trocar para TYPE_POSTPAID

File: system/library/mundipagg/src/Controller/Recurrence/Templates.php [hunk 3]
```diff
@@ -20,6 +20,12 @@ public function __call($name, array $arguments)
 
     public function index()
     {
+
```
Comment (michelpl) [inline system/library/mundipagg/src/Controller/Recurrence/Templates.php]: Linha extra

File: system/library/mundipagg/src/Factories/TemplateRootFactory.php [hunk 4]
```diff
@@ -16,33 +16,71 @@ class TemplateRootFactory
...
-        $templateRoot = new TemplateRoot();
         $templateRoot
             ->setTemplate($templateEntityFactory->createFromPostData($postData))
             ->setDueAt($dueAt)
-            ->setRepetitions($repetitions)
         ;
         return $templateRoot;
     }
+
+    public function createFromDBData($dbData) {
```
Comment (michelpl) [inline system/library/mundipagg/src/Factories/TemplateRootFactory.php]: PSR

File: system/library/mundipagg/src/Repositories/TemplateRepository.php [hunk 5]
```diff
@@ -22,8 +23,8 @@ protected function create(&$templateRoot)
                 `description`,
                 `accept_credit_card`,
                 `accept_boleto`,
-                `allow_installments`,                
-                `cycles`,
+                `allow_installments`,               
+
```
Comment (michelpl) [inline system/library/mundipagg/src/Repositories/TemplateRepository.php]: Linha extra

**Pull request merged**